### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Gmanlove**
## Completed tasks

- [ ] Write unit tests using Rspec.
- [ ] Add unit tests for all the methods in all the classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb
Comment (Kamba56) [inline spec/book_spec.rb:1]: - [ ] As demanded by project requirements, kindly add unit tests for each of the methods in all the classes. FOr the book class, unit tests for the `add_rental` and `to_h` are missing. Kindly add them please.

File: spec/classroom_spec.rb
Comment (Kamba56) [inline spec/classroom_spec.rb:1]: - [ ] As demanded by project requirements, kindly add unit tests for each of the methods in all the classes. For the classroom class, unit tests for the `add_student` is missing.

File: spec/person_spec.rb
Comment (Kamba56) [inline spec/person_spec.rb:1]: - [ ] As demanded by project requirements, kindly add unit tests for each of the methods in all the classes. For the person class, unit tests for the `add_rental`, `can_use_services`, and `corret_name` are missing.

File: spec/student_spec.rb
Comment (Kamba56) [inline spec/student_spec.rb:1]: - [ ] As demanded by project requirements, kindly add unit tests for each of the methods in all the classes. For the student class, unit tests for the `play_hooky` is missing.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Gmanlove , @Leeoasis, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job using rspec for unit tests!👏👏👏
- [x] Great job grouping tests using describe.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ] Consider adding details of the second author under the authors' section in the readme file.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'should return a Book object' do
+    expect(@book).to be_an_instance_of Book
+  end
+
+  describe '#title' do
+    it 'should return a book title' do
+      expect(@book.title).to eql 'Title'
+    end
```
Comment (ghost) [inline spec/book_spec.rb]: - [ ] Good work up to this point, one method in your book class is not tested yet ,`add_rental`  Kindly add the test as requested earlier

**Review by ghost — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ All tests are passing👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lizdev-05 — APPROVED** [review]
Hi @Gmanlove, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] Correct use of Gitflow 👍 
- [ ]  No linter error 👍 
- [ ]  All tests are passing👍
- [ ]  Great job using rspec for unit tests!👏👏👏
- [ ]  Great job grouping tests using describe.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**